test/unit/core.js

```diff
@@ -967,6 +967,25 @@ QUnit.test( "jQuery.grep()", function( assert ) {
 	assert.deepEqual( jQuery.grep( [ 1, 3, 5, 7 ], searchCriterion, false ), [], "Satisfying elements absent and grep explicitly uninverted" );
 } );
 
+QUnit.test( "jQuery.grep(Array-like)", function( assert ) {
+	assert.expect( 7 );
+
+	var searchCriterion = function( value ) {
+		return value % 2 === 0;
+	};
+
+	assert.deepEqual( jQuery.grep( { length: 0 }, searchCriterion ), [], "Empty array-like" );
+
+	assert.deepEqual( jQuery.grep( { 0: 1, 1: 2, 2: 3, 3: 4, 4: 5, 5: 6, length: 6 }, searchCriterion ), [ 2, 4, 6 ], "Satisfying elements present and array-like object used" );
```

Comment:
@markelog Done. Please, let me know if that's OK. Also, you don't have to apologize. It's totally fine to do more work on the PR to improve its quality standard.